.github/workflows/nightly_cleanup.yml

```diff
@@ -0,0 +1,80 @@
+---
+name: Nightly Cleanup
+
+on:
+  schedule:
+    - cron: '0 5 * * *'
+  workflow_dispatch:
+
+
+env:
+  AWS_PROFILE: "infex"
+  AWS_REGION: "eu-central-1"
+
+jobs:
+  aws-nightly-cleanup:
+    runs-on: ubuntu-latest
+    # Other dependencies from hosted runner
+    # AWS CLI
+    # indirectly node used by actions
+
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+
+      - name: Import Secrets
+        id: secrets
+        uses: hashicorp/vault-action@d1720f055e0635fd932a1d2a48f87a666a57906c # v3
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          exportEnv: false
+          secrets: |
+            secret/data/products/infrastructure-experience/ci/common AWS_ACCESS_KEY;
+            secret/data/products/infrastructure-experience/ci/common AWS_SECRET_KEY;
+
+      # Official action does not support profiles
+      - name: Add profile credentials to ~/.aws/credentials
+        run: |
+          aws configure set aws_access_key_id ${{ steps.secrets.outputs.AWS_ACCESS_KEY }} --profile ${{ env.AWS_PROFILE }}
+          aws configure set aws_secret_access_key ${{ steps.secrets.outputs.AWS_SECRET_KEY }} --profile ${{ env.AWS_PROFILE }}
+          aws configure set region ${{ env.AWS_REGION }} --profile ${{ env.AWS_PROFILE }}
+
+      - name: Install Cloud Nuke
+        run: |
+          wget https://github.com/gruntwork-io/cloud-nuke/releases/download/v0.33.0/cloud-nuke_linux_amd64
+          chmod +x cloud-nuke_linux_amd64
+
+      # This is likely to fail, therefore we ignore the error
+      # We're ignoring ec2_dhcp_option as they couldn't be deleted
+      # cloudtrail is managed by IT and can't be deleted either
+      - name: Run Cloud Nuke
+        timeout-minutes: 45
+        env:
+          DISABLE_TELEMETRY: "true"
+        run: |
+          ./cloud-nuke_linux_amd64 aws \
+          --region ${{ env.AWS_REGION }} \
+          --force \
+          --newer-than 4h \
```

Comment:
I'd go with a slightly higher number like 6 hours here and line 78.
It's not guaranteed that your GitHub Action Job is scheduled on point and 4 hours is the exact time difference between the two jobs, so might run into an unforeseen edge case that it may not clean up everything.